File: Examples/VisualBasic/Tables/ExtractTableText.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic
Namespace Tables
    Public Class ExtractTableText
        Public Shared Sub Run()
            ' ExStart:ExtractTableText
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Tables()

            ' Load the document into Aspose.Note.
            Dim document As New Document(dataDir & Convert.ToString("Sample1.one"))

            ' Get a list of table nodes
            Dim nodes As IList(Of Table) = document.GetChildNodes(Of Table)()

            ' Set table count
            Dim tblCount As Integer = 0

            For Each table As Table In nodes
                tblCount += 1
                Console.WriteLine("table # " + tblCount.ToString())

                ' Retrieve text
                Dim text As String = table.GetText()
                ' Print text on the output screen
                Console.WriteLine(text)
            Next
            ' ExEnd:ExtractTableText
        End Sub
    End Class
End Namespace

File: Examples/CSharp/Tables/ExtractTableText.cs
using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Aspose.Note.Examples.CSharp.Tables
{
    public class ExtractTableText
    {
        public static void Run()
        {
            // ExStart:ExtractTableText
            // ExFor:Table
            // ExFor:RichText
            // ExFor:RichText.Text
            // ExSummary:Shows how to get text from a table.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Tables();

            // Load the document into Aspose.Note.
            Document document = new Document(dataDir + "Sample1.one");

            // Get a list of table nodes
            IList<Table> nodes = document.GetChildNodes<Table>();

            // Set table count
            int tblCount = 0;

            foreach (Table table in nodes)
            {
                tblCount++;
                Console.WriteLine("table # " + tblCount);

                // Retrieve text
                string text = string.Join(Environment.NewLine, table.GetChildNodes<RichText>().Select(e => e.Text)) + Environment.NewLine;
               
                // Print text on the output screen
                Console.WriteLine(text);
            }

            // ExEnd:ExtractTableText
        }
    }
}